hera_cal/reflections.py

```diff
@@ -272,11 +271,15 @@ def model_auto_reflections(self, clean_resid, dly_range, clean_flags=None, clean
 
         # form gains
         self.ref_gains = form_gains(self.ref_eps)
-        for a in self.ants:
-            for p in self.pols:
-                k = (a, uvutils.parse_jpolstr(p)) 
-                if k not in self.ref_gains:
-                    self.ref_gains[k] = np.ones_like(clean_resid[keys[0]], dtype=np.complex)
+
+        # if ref_gains not empty, fill in missing antenna and polarizations with unity gains
+        if len(self.ref_gains) > 0:
+            _keys = list(self.ref_gains.keys())
+            for a in self.ants:
+                for p in self.pols:
+                    k = (a, uvutils.parse_jpolstr(p))
```

Comment:
`self.pols` is visibility polarizations, no? Shouldn't you use `hera_cal.utils.split_pol(p)[0]`?